packages/grid/src/vaadin-grid-mixin.js

```diff
@@ -418,8 +418,19 @@ export const GridMixin = (superClass) =>
         })
         .forEach((cell) => {
           cell.__measuringAutoWidth = autoWidth;
-          cell._content.style.width = autoWidth ? 'auto' : '';
-          cell._content.style.position = autoWidth ? 'absolute' : '';
+
+          if (cell.__measuringAutoWidth) {
+            // Store the original inline width of the cell to restore it later
+            cell.__originalWidth = cell.style.width;
+            // Prepare the cell for having its intrinsic width measured
+            cell.style.width = 'auto';
+            cell.style.position = 'absolute';
+          } else {
+            // Restore the original width
+            cell.style.width = cell.__originalWidth || '';
+            delete cell.__originalWidth;
+            cell.style.position = '';
+          }
```

Comment:
Modifying the inline style `width` of a cell requires additional logic to have it restored to its original value after measuring. This is the main reason the cell content was originally used.